src/orion/core/cli/hunt.py

```diff
@@ -29,9 +30,18 @@ def add_subparser(parser):
 
     orion_group.add_argument(
         '--max-trials', type=int, metavar='#',
-        help="number of jobs/trials to be completed "
+        help="number of trials to be completed for the experiment. This value "
+             "will be saved within the experiment configuration and reused "
+             "across all workers to determine experiment's completion. "
              "(default: %s)" % resolve_config.DEF_CMD_MAX_TRIALS[1])
 
+    orion_group.add_argument(
+        '--worker-trials', type=int, metavar='#',
+        help="number of trials to be completed for this worker. "
+             "If the experiment is completed, the worker will die even if it "
+             "did not reach its maximum number of trials "
+             "(default: %s)" % resolve_config.DEF_CMD_WORKER_TRIALS[1])
```

Comment:
Thinking about this, this will get a terrible rendering; ```(default: 9223372036854775807)```. I would rather stick to ```numpy.inf``` and handle it at the for-loop in ```worker```. Sorry for changing my mind late. :sweat_smile:

Reply:
It renders `default: inf/until preempted`

Reply:
`numpy.inf` is float, and even when you do `int(numpy.inf)` it crashes. It is very inconvenient value.

Reply:
Oh, that's right. :thinking: I'm still not very comfortable with that fact that it is a precise value that cannot be said to be inf. For instance 1000000000000 would have the same practical effect but we could not say it is inf. But whatever, I guess that may not be very important.

Reply:
Sometimes people use -1 for infinity, but this is also not very nice, because you have to do more coding with if/else everywhere.

Reply:
Yeah, so would 'inf'. But both would not generate situations where the user load back its configuration and see ```{'max_trials': 9223372036854775807}``` :astonished:. I don't mind a couple of if-else if it makes user experience better. :wink: Also, -1 or 'inf' are not likely to change from one platform to another.

Reply:
What about using `None` for infinity?

Reply:
Could also be. What would be the advantage against `numpy.inf`?

Reply:
Advantage is that it is a special value. Some integer functions work with `numpy.inf`, some don't. In the contrast, most of things will crash when `None` is passed. So, it will be easier to debug in the future. ("Let it crash" philosophy I guess)

Reply:
And I really don't like the fact that variables defined with `int` type are assigned `float` type.

Reply:
Those are good points indeed. I'm not sure however what could someone try to do with `worker_trials` other than printing it or iterating on it. So printing would work either way, and iterating would crash either way. `numpy.inf` however has the nice attribute that it is comparable with the proper behavior we are looking for.

Reply:
What about `max_trials`? Where is it used?

Reply:
There for instance:
https://github.com/mila-udem/orion/blob/master/src/orion/core/worker/experiment.py#L327